inst/scripts/SEL3D.R

```diff
@@ -0,0 +1,139 @@
+    
+    if (length(tax) > 0) {
+      blastOutput$tax[entry] <- tax
+    } else {
+      blastOutput$tax[entry] <- NA
+    }
+    
+    if(length(info) > 0) {
+      blastOutput$info[entry] <- info
+    } else {
+      blastOutput$info[entry] <- NA
+    }
+
+    
+    if(!intersectingPositions(finalOutputRange, blastOutput$q.start[entry], blastOutput$q.end[entry])){
+    # Finally, we then find the best match that is either Homo sapiens or 100% ID.
+      if((!is.na(blastOutput$tax[entry]) && blastOutput$tax[entry] == "Homo sapiens") || blastOutput$identity[entry] == 100){
+        finalOutput <- rbind(finalOutput, blastOutput[entry,])
+        finalOutputRange[[finalOutputRangePointer]] <- c(blastOutput$q.start[entry]:blastOutput$q.end[entry])
+        finalOutputRangePointer <- finalOutputRangePointer + 1
+      }
+    }
+  }
+  
+  #no matches 
+  if(nrow(finalOutput) == 0){ 
+    message(sprintf("all %d matches were dropped, returning NULL", nrow(blastOutput)))  
+    return(NULL) 
+  }
+  
+  finalOutput$q.start <- finalOutput$q.start + startPos - 1
+  finalOutput$q.end <- finalOutput$q.end + startPos - 1
+  
+  for(x in 1:length(finalOutputRange)){
+    finalOutputRange[[x]] <- finalOutputRange[[x]] + startPos - 1
+  }
+  
+  minSide <- min(finalOutputRange[[1]])
+  maxSide <- max(finalOutputRange[[length(finalOutputRange)]])
+  
+  if(endPos - maxSide > 50){
+    message(sprintf("running substr from pos %d to %d", maxSide, endPos))
+    maxFrame <- getPdbIds(substr(AAseq, maxSide - startPos, endPos - startPos), 
+                             startPos = maxSide, endPos = endPos, convert = FALSE, silent = silent)
+    finalOutput <- rbind(finalOutput, maxFrame)
+  }
+  
+  if(minSide > 50){
+    message(sprintf("running substr from pos %d to %d", startPos, minSide))
+    minFrame <- getPdbIds(substr(AAseq, 1, minSide - startPos), 
+                             startPos = startPos, endPos = minSide, convert = FALSE, silent = silent)
+    finalOutput <- rbind(finalOutput, minFrame)
+  }
+  
+  for(y in 1:(length(finalOutputRange) - 1)){
+    leftSide <- max(finalOutputRange[[y]])
+    rightSide <- min(finalOutputRange[[y+1]])
+    
+    if(rightSide - leftSide > 50){
+      message(sprintf("running substr from pos %d to %d", leftSide, rightSide))
+      interFrame <- getPdbIds(substr(AAseq, leftSide - startPos, rightSide - startPos), 
+                               startPos = leftSide, endPos = rightSide, convert = FALSE, silent = silent)
+      finalOutput <- rbind(finalOutput, interFrame)
+    }
+  }
+  
+  return(finalOutput)
+}
+#We need to store ranges of the best alignments 
+#(as start (st) and end (ed) positions) so we can
+#add that match to the final output without adding
+#segments which overlap in the final format.
+intersectingPositions <- function(r, st, ed){
+  for(x in 1:length(r)){
+    if(length(intersect(r[[x]], c(st:ed))) > 0){
+      return(TRUE)
+    }
+  }
+  return(FALSE)
+}
```

Comment:
Yes they are. You were supposed to write code after writing Unit Test cases. Since we are following a test driven development process, the tests you write should define the desired function of your code. Since you wrote code, I expected unit tests to be here as well. Each of my ToDo comments were steps you skipped in the process.